pkg/apis/kappctrl/v1alpha1/types.go

```diff
@@ -9,15 +9,22 @@ import (
 
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name=Description,JSONPath=.status.friendlyDescription,description=Friendly description,type=string
+// +kubebuilder:printcolumn:name=Since-Deploy,JSONPath=.status.deploy.startedAt,description=Last time app started being deployed. Does not mean anything was changed.,type=date
+// +kubebuilder:printcolumn:name=Age,JSONPath=.metadata.creationTimestamp,description=Time since creation,type=date
 type App struct {
 	metav1.TypeMeta `json:",inline"`
 
 	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata.
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   AppSpec   `json:"spec"`
-	Status AppStatus `json:"status"`
+	// +optional
+	Spec AppSpec `json:"spec,omitempty"`
```

Comment:
Is AppSpec optional?

Reply:
Not sure what best practice is here and was wondering myself. I think what you are suggesting makes sense, but I mainly did this because I was following what other projects had done (e.g. velero, knative).

Reply:
Made spec required